### PR TITLE
Update to Angular 6

### PR DESCRIPTION
**Pull request by timdeschryver**

### CODE REVIEW

**Comment by timdeschryver** [comment]
The latest commit uses the angular cli to build the library.
Feedback is as always welcome, I'll walk through it again tomorrow.

File: projects/ngx-drag-to-select/ng-package.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
+  "dest": "../../dist/ngx-drag-to-select",
+  "deleteDestPath": false,
+  "lib": {
+    "entryFile": "src/public_api.ts",
+    "styleIncludePaths": ["./src/scss"]
```
Comment (timdeschryver) [inline projects/ngx-drag-to-select/ng-package.json]: I tried to build the styles with the cli, but the styles didn't get picked up.
Any ideas? Or do we keep it as it is?

File: projects/ngx-drag-to-select/tslint.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "../../tslint.json",
+  "rules": {
+    "directive-selector": [false, "attribute", "ngx", "camelCase"],
```
Comment (timdeschryver) [inline projects/ngx-drag-to-select/tslint.json]: This rule is disabled for now. 
When we rename `selectItem` to something more suitable, we'll have to turn this back on.

File: src/app/app.module.ts [hunk 3]
```diff
@@ -29,6 +28,7 @@ import { FooterComponent } from './footer/footer.component';
 import { HeaderComponent } from './header/header.component';
 import { PhoneComponent } from './phone/phone.component';
 import { LayoutModule } from '@angular/cdk/layout';
+import { DragToSelectModule } from '../../projects/ngx-drag-to-select/src/lib/drag-to-select.module';
```
Comment (timdeschryver) [inline src/app/app.module.ts]: Do we want to import from `ngx-drag-to-select` here?
If so I think we'll have to modify the build process I think.

File: package.json [hunk 4]
```diff
@@ -11,14 +11,14 @@
...
-    "build:server": "ng build --prod --app 1 --output-hashing=false",
-    "build:lib": "npm run clean && npm run packagr && npm run build:lib:sass && npm run copy:styles && npm run copy:readme",
-    "build:lib:sass": "node-sass src/lib/scss/ngx-drag-to-select.scss -o dist/lib",
-    "packagr": "ng-packagr -p src/lib/package.json",
+    "build:app": "ng build --prod --project=app",
+    "build:server": "ng build --prod --project=server",
+    "build:lib": "npm run clean && npm run build:lib:code && npm run build:lib:sass && npm run copy:styles && npm run copy:readme",
+    "build:lib:code": "ng build --prod --project=ngx-drag-to-select",
+    "build:lib:sass": "node-sass projects/ngx-drag-to-select/src/scss/ngx-drag-to-select.scss -o dist/ngx-drag-to-select",
     "deploy": "ngh --dir dist/browser",
```
Comment (timdeschryver) [inline package.json]: Is this used @d3lm ?

File: projects/ngx-drag-to-select/src/lib/operators.ts [hunk 5]
```diff
@@ -1,14 +1,16 @@
...
 
 export const createSelectBox = (container: SelectContainerHost) => (
   source: Observable<SelectBoxInput>
 ): Observable<SelectBox<number>> =>
   source.pipe(
     map(([event, opacity, { x, y }]) => {
-      const mousePosition = getRelativeMousePosition(event, container);
+      // explicit cast to MousePosition in order to "use" `MousePosition`
+      // we have to import `MousePosition` otherwise we'll get a build error
+      const mousePosition: MousePosition = getRelativeMousePosition(event, container);
```
Comment (timdeschryver) [inline projects/ngx-drag-to-select/src/lib/operators.ts]: Do you got a better solution? 
The build was throwing an error because it didn't know `MousePosition`

File: projects/ngx-drag-to-select/src/lib/operators.ts [hunk 6]
```diff
@@ -1,14 +1,16 @@
...
 
 export const createSelectBox = (container: SelectContainerHost) => (
   source: Observable<SelectBoxInput>
 ): Observable<SelectBox<number>> =>
   source.pipe(
     map(([event, opacity, { x, y }]) => {
-      const mousePosition = getRelativeMousePosition(event, container);
+      // Explicit cast to `MousePosition` in order to "use" `MousePosition`
+      // we have to import `MousePosition` otherwise we'll get a build error
+      const mousePosition: MousePosition = getRelativeMousePosition(event, container);
```
Comment (timdeschryver) [inline projects/ngx-drag-to-select/src/lib/operators.ts]: Do you got a better solution?
The build was throwing an error because it didn't know `MousePosition`.

File: package.json (hunk 4 above)
Comment (d3lm) [reply package.json]: This is only meant to be used locally, if needed.

File: projects/ngx-drag-to-select/src/lib/operators.ts (hunk 6 above)
Comment (d3lm) [reply projects/ngx-drag-to-select/src/lib/operators.ts]: And this doesn't work if we add the return type to `getRelativeMousePosition`?

File: projects/ngx-drag-to-select/ng-package.json (hunk 1 above)
Comment (d3lm) [reply projects/ngx-drag-to-select/ng-package.json]: I'll check out the PR later on my local machine and see if I find something.

Comment (timdeschryver) [reply projects/ngx-drag-to-select/ng-package.json]: You should also add ` "styleIncludePaths": ["./src/scss"]` to `ng-package-prod.json`(?) file because we're building it with the `prod` flag.

File: projects/ngx-drag-to-select/src/lib/operators.ts (hunk 6 above)
Comment (timdeschryver) [reply projects/ngx-drag-to-select/src/lib/operators.ts]: I'll try it out later today.

Comment (timdeschryver) [reply projects/ngx-drag-to-select/src/lib/operators.ts]: I did a quick check and it doesn't work.
I added a return type to `getMousePosition` because `getRelativeMousePosition` already has a return type?

File: package.json [hunk 7]
```diff
@@ -7,29 +7,29 @@
...
     "style": "npm run lint:check && npm run format:check",
     "style:fix": "npm run lint:fix && npm run format:fix",
     "clean": "del dist",
-    "test": "jest",
+    "test": "jest --config ./projects/ngx-drag-to-select/jest.config.js",
     "test:ci": "npm run test && start-server-and-test start http-get://localhost:4200 e2e:ci",
-    "test:watch": "jest --watch",
+    "test:watch": "npm run test -- --watch",
     "lint:check": "ng lint",
     "lint:fix": "ng lint --fix",
```
Comment (d3lm) [inline package.json]: This command is broken with the latest CLI. It doesn't allow multiple apps for a single command I suppose. We have to change this to `ng lint app --fix && ng lint ngx-drag-to-select --fix`.

Comment (timdeschryver) [reply package.json]: Ah good catch, you're right.

Comment (timdeschryver) [reply package.json]: Done!

File: projects/ngx-drag-to-select/src/lib/operators.ts (hunk 6 above)
Comment (d3lm) [reply projects/ngx-drag-to-select/src/lib/operators.ts]: Yep, indeed. This makes sense. The compiler is trying to figure out the shape of `MouseEvent` but it can't cause it's defined in a different module and never imported to the `operators.ts` module. So we can just add the type annotation and remove the comment I guess. What do you think? Or should we keep the comment?

File: package.json (hunk 7 above)
Comment (d3lm) [reply package.json]: Do you need the `--project` flag?

File: projects/ngx-drag-to-select/src/lib/operators.ts (hunk 6 above)
Comment (timdeschryver) [reply projects/ngx-drag-to-select/src/lib/operators.ts]: I'm afraid future us will try to remove it 😅 and bumping into this problem again.

File: package.json (hunk 7 above)
Comment (timdeschryver) [reply package.json]: Nope we don't! I also removed the `--project` flag in the build scripts.

File: projects/ngx-drag-to-select/src/lib/operators.ts (hunk 6 above)
Comment (d3lm) [reply projects/ngx-drag-to-select/src/lib/operators.ts]: Haha you're right, we can leave it in then. Maybe let's reword this to `Type annotation is required here, because it implicitly uses MousePosition can the TS compiler has no idea what about the shape of this type.`, or something along those lines 🙂

File: package.json (hunk 7 above)
Comment (d3lm) [reply package.json]: Yay 🎉

**Comment by d3lm** [comment]
Woho 🎉 We finally made it to Angular 6. Thanks so much @timdeschryver for all the work you put into this PR. I really appreciate your help.
